ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/datetime/FormatDateTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.datetime;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.fail;
+
+import io.confluent.ksql.function.KsqlFunctionException;
+import java.sql.Date;
+import java.util.stream.IntStream;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FormatDateTest {
+
+  private FormatDate udf;
+
+  @Before
+  public void setUp() {
+    udf = new FormatDate();
+  }
+
+  @Test
+  public void shouldConvertDateToString() {
+    // When:
+    final String result = udf.formatDate(Date.valueOf("2014-11-09"), "yyyy-MM-dd");
+
+    // Then:
+    assertThat(result, is("2014-11-09"));
+  }
+
+  @Test
+  public void shouldRoundTripWithStringToDate() {
+    final String format = "dd/MM/yyyy'Freya'";
+    final ParseDate parseDate = new ParseDate();
+    IntStream.range(-10_000, 20_000)
+        .parallel()
+        .forEach(idx -> {
+          final String result = udf.formatDate(new Date(idx * 86400000L), format);
+          final Date date = parseDate.parseDate(result, format);
+          assertThat(date.getTime(), is(idx * 86400000L));
+        });
+  }
+
+  @Test
+  public void shouldSupportEmbeddedChars() {
+    // When:
+    final Object result = udf.formatDate(Date.valueOf("2014-11-09"), "yyyy-dd-MM'Fred'");
+
+    // Then:
+    assertThat(result, is("2014-09-11Fred"));
```

Comment:
I'm not sure about this. Should we support embedded chars in the final date string?

Reply:
It was supported in the old DateToString function: https://github.com/confluentinc/ksql/blob/cd1a9880aa39e0fb9b4e84d9f713235c8569b104/ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/datetime/DateToStringTest.java#L61

Reply:
Just tested Mysql and it supports this too. Perhaps it is allowed in order to write the format you want, i.e. dates `yyyy/MM/dd` or `yyyy-MM-dd` or `Today is Year = yyyy Month = MM Day = dd`